frame/node-permission/Cargo.toml

```diff
@@ -0,0 +1,35 @@
+[package]
+name = "pallet-node-permission"
```

Comment:
Maybe a better name?

@danforbes any ideas?

Reply:
Is this pallet generic for all types of nodes, or can we use a more specific word, like validator, author or authority? I think the term `permission` is fine; possible improvement could be `node-permission-manager` or, if it makes more sense `author-permission-manager`.

Reply:
Other terms to keep in mind that may help: "access control" and "authorization".

Reply:
It looks like the term "peer" is used elsewhere in this PR, would it make sense to use that term here? `peer-permission-manager`?

Reply:
Peer is mostly used in network layer, I not sure if it makes sense to use it in runtime, maybe @tomaka has some insights? An extra `manager` sounds redundant, like we are making staking -> staking-manager.

Reply:
Subjectively, I don't know if I agree with your analogy - I find "staking" to be a much more specific term than "permission". I agree with Basti that the name of this pallet can be improved by making it more specific and evocative. I would suggest clarifying the name to more narrowly define what "node" means in this case as well as to indicate that the pallet does not just _define_ permissions, it also exposes an interface to _manage_ them, which is a rather different concern I would say. I think `pallet-<author/validator>-access-control` may be a good name.

Reply:
Is node/peer-authentication a better name or even worse?

Reply:
I would say worse because this is _authorization_ not _authentication_.

Reply:
If this system is really generic across _all_ types of nodes, then I think `pallet-node-authorization` is a very nice name. I would really push for clarifying `node` as much as possible, though. It's a very ambiguous term.

Reply:
@danforbes thanks for the insights, I now agree with you that `node` is too generic for this specific usage, and I like authorization, I would like take `pallet-peer-authorization` if no objections.

Reply:
Sounds great to me! Can you explain a little bit why you like the word "peer" in this context? I'm not very familiar with the technical use of this term when it comes to Substrate. I imagine it has a bit to do with networking, but is there anything beyond that I should keep in mind?

Reply:
From my understanding (can't ensure it's fully correct 😄 )  node is an abstraction includes the fully software like network, runtime and many others; peer in Substrate mostly exists in network since we are using libp2p which has a specific definition for [Peer](https://github.com/libp2p/rust-libp2p/blob/master/core/src/network/peer.rs#L57).
It's still interchangeably in most chat, check [this](https://docs.libp2p.io/reference/glossary/#node).
But for our use case `peer` is more accurate and less ambiguous.

Reply:
During struggling commit the new name, I failed to persuade myself to use `peer` again when having to rename dispatch calls with `add_peer` and type with `PeerId` (confused with the same type in libp2p). `pallet-node-authorization` is the name then, other improvements are appreciate.

Reply:
A "peer" is a node we're connected to.